### ISSUE TITLE
Login regularly fails (20.3.7)

### ISSUE THREAD

**Issue by madsci** [rbuehlma/pvr.zattoo#173]
Hi rbuehlma

As stated by jonas99g (https://github.com/rbuehlma/pvr.zattoo/issues/172#issue-1679152060), I have also regular login issues with LE 11.0.1 on a RPi4. (Multiple) restarting of the PVR addon helps.

Full debug log is here: http://ix.io/4vTr

For me it looks like an issue with zapi/v3/account/login:
2023-05-14 12:53:07.370 T:14005 debug : UpdateComponents: Skipping not (yet) connected PVR client 'pvr.zattoo'
2023-05-14 12:53:07.382 T:14004 debug : AddOnLog: pvr.zattoo: Got cookie: beaker.session.id.
2023-05-14 12:53:07.382 T:14004 debug : AddOnLog: pvr.zattoo: Got new beaker.session.id: a3f6c..
2023-05-14 12:53:07.382 T:14004 debug : AddOnLog: pvr.zattoo: Need to login.
2023-05-14 12:53:07.382 T:14004 debug : AddOnLog: pvr.zattoo: Try to login.
2023-05-14 12:53:07.382 T:14004 debug : AddOnLog: pvr.zattoo: Http-Request: POST https://zattoo.com/zapi/v3/account/login.
2023-05-14 12:53:07.382 T:14004 debug : CurlFile::Open - <https://zattoo.com/zapi/v3/account/login>
2023-05-14 12:53:07.401 T:14004 error : CCurlFile::Open - <https://zattoo.com/zapi/v3/account/login> Failed with code 0:
2023-05-14 12:53:07.403 T:14004 error : AddOnLog: pvr.zattoo: Open URL failed with -2.
2023-05-14 12:53:07.403 T:14004 error : AddOnLog: pvr.zattoo: Login failed.
2023-05-14 12:53:07.403 T:14004 error : AddOnLog: pvr.zattoo: Login failed
2023-05-14 12:53:07.484 T:14005 debug : UpdateComponents: Skipping not (yet) connected PVR client 'pvr.zattoo'
2023-05-14 12:53:08.485 T:14005 debug : UpdateComponents: Skipping not (yet) connected PVR client 'pvr.zattoo'

When I try to access account/login directly with a browser (after login at webinterface), following error is shown:
{"http_status":404,"internal_code":1,"message":"Invalid path or HTTP method","success":false}

Thank you

**Comment by rbuehlma** [comment]
I have observed the same behaviour, unfortunately, I have no Idea what the cause is. After some restarts, it starts working again. It does not seem to be an issue with the addon but due to some unknown issue, the addon does not receive any data from the server. This could be an issue with the CURL or TLS library which gets triggered by some changes on server side by Zattoo but not sure about this.

**Comment by italfire** [comment]
same problem here with Kodi Matrix and Nexus. sometimes it works on the first try other times I have to restart kodi several times

**Comment by vinz1080** [comment]
Same issue. Random restart of Kodi Nexus (Coreelec 20) using pvr.zattoo 20.3.7. After these Kodi crashes I always have pvr.zattoo invalid credential error. I have to restart the plugin one or multiple times to finally success. Really annoying :(

**Comment by rbuehlma** [comment]
The random restarts most probably are a different problem. The cause for the login issue is not yet clear.

**Comment by madsci** [comment]
> I have observed the same behaviour, unfortunately, I have no Idea what the cause is. After some restarts, it starts working again. It does not seem to be an issue with the addon but due to some unknown issue, the addon does not receive any data from the server. This could be an issue with the CURL or TLS library which gets triggered by some changes on server side by Zattoo but not sure about this.

Interestingly, when I try to fetch https://zattoo.com/zapi/v3/account/login with curl, in two-out-of-three or every second time, a timeout occurs. Could it be a workaround to trigger the login several times in a row to get a good response?

Furthermore, I decompiled the android app to find hints to the login-procedure and urls. In a first scan, the url and parameters seems to be correct. However, I cannot tell if for instance the order of the init process is still ok.
```
sources\com\zattoo\zsessionmanager\internal\network\C18104b.java
@C21358o("zapi/v3/account/login")
@C21348e
/* renamed from: e */
C19683w<C18113g> mo56474e(@C21346c("login") String str, @C21346c("password") String str2, @C21346c("remember") String str3);
```


It could also be that zattoo is forcing newer login methods. The api obviously supports other login methods:
```
/* renamed from: b */
C19683w<C18113g> mo56471b(@C21346c("external_token") String str, @C21346c("remember") String str2);

/* renamed from: f */
C19683w<C18113g> mo56475f(@C21362s("ssoProviderId") String str, @C21346c("access_token") String str2, @C21346c("remember") String str3);
```

**Comment by italfire** [comment]
I just noticed that if I reinstall Zatto PVR (without deleting the installed version) immediately loads the list without needing to restart the device. This method works every time so I thought it could be useful information to be able to solve the problem

**Comment by rbuehlma** [comment]
Hopefully this is fixed or at least improved with 20.3.8

**Comment by p0nti** [comment]
How do i get 20.3.8 ? On libreelec it doesn't update automatically (so still stuck with 20.3.7) and when i want to install 
 the latest version from https://kodi.tv/addons/nexus/pvr.zattoo/  and the chose "install from zip" the installation always fails because of "invalid file structure".

**Comment by madsci** [comment]
It indeed seems that le is slower than other distros in that subject. But 20.3.8 is available in the meantime. Try to search for updates under user's addons -> side menu on the left, and install it in the zattoo addon prefs under versions.

**Comment by vinz1080** [comment]
> The random restarts most probably are a different problem. The cause for the login issue is not yet clear.

pvr.zattoo v20.3.8 actually resolved my login issue. Thank you very much for the fix. Regarding crashes I had, I finally identified the source. Inputstream.adaptive v20.3.7 has a memory leak issue. It is fixed in v20.3.8. https://github.com/xbmc/inputstream.adaptive/releases/tag/20.3.8-Nexus
